app/workers/edx_course_worker.rb

```diff
@@ -20,92 +32,93 @@ def handle_response_data(response_data)
     xseries_track_type = CourseTrackType.find_by(type_of_achievement: 'edx_xseries_verified_certificate')
     profed_track_type = CourseTrackType.find_by(type_of_achievement: 'edx_profed_certificate')
 
-    response_data['value']['items'].each do |course_element|
-      course = Course.find_by(provider_course_id: course_element['course:id'], mooc_provider_id: mooc_provider.id)
-      if course.nil?
-        course = Course.new
-      else
-        update_map[course.id] = true
-      end
+    response_data.each do |xml_doc|
+      xml_doc.xpath('//channel/item').each do |course_element|
+        course = Course.find_by(provider_course_id: course_element.xpath('course:id').text, mooc_provider_id: mooc_provider.id)
```

Comment:
What about our special method `Course.get_course_id_by_mooc_provider_id_and_provider_course_id`? You don't have to use it, just if you like.